pkg/chains/storage/docdb/docdb.go

```diff
@@ -327,3 +341,58 @@ func getMongoServerURLFromDir(dir string) (string, error) {
 
 	return fileDataNormalized, nil
 }
+
+// getMongoServerUrlFromPath retreives token from the given mount path
+func getMongoServerURLFromPath(path string) (string, error) {
+	fileData, err := os.ReadFile(path)
+	if err != nil {
+		return "", fmt.Errorf("reading file in %q: %w", path, err)
+	}
+
+	// A trailing newline is fairly common in mounted files, so remove it.
+	fileDataNormalized := strings.TrimSuffix(string(fileData), "\n")
+	return fileDataNormalized, nil
+}
+
+func getPathsToWatch(ctx context.Context, cfg config.Config) []string {
+	var pathsToWatch = []string{}
+	logger := logging.FromContext(ctx)
+
+	if cfg.Storage.DocDB.MongoServerURLPath != "" {
+		logger.Infof("setting up fsnotify watcher for mongo server url path: %s", cfg.Storage.DocDB.MongoServerURLPath)
+		dirPath := filepath.Dir(cfg.Storage.DocDB.MongoServerURLPath)
+		pathsToWatch = []string{
+			// mongo-server-url-path/<path> is where the mongo server url token
+			// When a Kubernetes secret is mounted on a path, the `data` in that secret is mounted
+			// under path/..data that is then `symlink`ed to the key of the data. In this instance,
+			// the mounted path is going to look like:
+			// file 1 - ..2024_05_03_11_23_23.1253599725
+			// file 2 - ..data -> ..2024_05_03_11_23_23.1253599725
+			// file 3 - ..data/<path>
+			// So each time the secret is updated, the file is not updated,
+			// instead the underlying symlink at `..data` is updated and that's what we want to
+			// capture via the fsnotify event watcher
+			// filepath.Join(cfg.Storage.DocDB.MongoServerURLPath, "..data"),
+			filepath.Join(dirPath, "..data"),
+		}
+	} else if cfg.Storage.DocDB.MongoServerURLDir != "" {
```

Comment:
Similar comment as a previous one. Let's make this an `if-block` instead of an `else-if-block`.